src/components/CallView/shared/VideoVue.vue

```diff
@@ -482,6 +482,19 @@ export default {
 	},
 
 	watch: {
+		model: {
+			immediate: true,
+
+			handler(model, oldModel) {
+				if (oldModel) {
+					oldModel.off('reaction', this._handleReaction)
+				}
+
+				if (model) {
+					model.on('reaction', this._handleReaction)
+				}
+			},
+		},
```

Comment:
If I am not mistaken adding a handle for the reaction event emitted by the model should not be done in the VideoView, unless it is desired that only reactions for visible participants are shown.

Moreover, I do not think that the props and hence the model are nullified when the component is destroyed (but I might be wrong, I am far from a Vue expert), so the event listener is not properly removed when the component is destroyed. This could explain why reactions multiply after switching from speaker mode to grid view and back again: new VideoView components are created, but although the older ones are destroyed the listeners are still running (this might happen already in the example code included in #9311, but I have not checked).

I would suggest to do something like passing [`callParticipantModels`](https://github.com/nextcloud/spreed/blob/c74f038d317ade3e20dd4493b7513733a46aa75a/src/components/CallView/CallView.vue#L207) to the `ReactionToaster` and watching it for changes, adding or removing a listener for each model as needed. Something similar to (**untested**, based on the (horrible :-P ) code of [`updateDataFromCallParticipantModels`](https://github.com/nextcloud/spreed/blob/c74f038d317ade3e20dd4493b7513733a46aa75a/src/components/CallView/CallView.vue#L438), so if you have any cleaner idea please do that instead ;-) ):
```
data: {
    registeredModels: {},
},
...
watch: {
    callParticipantModels(models) {
        const addedModels = models.filter(model => !this.registeredModels[model.attributes.peerId])
        const removedModelIds = Object.keys(this.registeredModels).filter(registeredModelId => models.find(model => model.attributes.peerId === registeredModelId) === undefined)

        removedModelIds.forEach(removedModelId => {
            this.registeredModels[addedModel.attributes.peerId].off('reaction', this._handleReaction)
            delete this.registeredModels[addedModel.attributes.peerId]
        })

        addedModels.forEach(addedModel => {
            this.registeredModels[addedModel.attributes.peerId] = addedModel
            this.registeredModels[addedModel.attributes.peerId].on('reaction', this._handleReaction)
        })
    },
},
methods: {
    _handleReaction(model, reaction) {
        ...
    },
},
```

Reply:
Thanks for the tip!